app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/amend/unmet_dependency.scala.html

```diff
@@ -32,11 +33,17 @@
         submitButton: uk.gov.hmrc.cgtpropertydisposalsfrontend.views.html.components.submit_button
 )
 
-@(backLink: Call, titleKey: String)(implicit messages:Messages, appConfig: ViewConfig, request: RequestWithSessionData[_])
+@(backLink: Call, section: String)(implicit messages:Messages, appConfig: ViewConfig, request: RequestWithSessionData[_])
```

Comment:
there's some logic in the controller which'll need updating as well for the right string to be passed in - at the moment it will give you the whole title message key, not just the part which tells you which question they were trying to change

Reply:
```scala
  private def unmetDependencyTitleKey(unmetDependencyFieldUrl: String): Option[String] = {
    def is(s: String): Boolean = unmetDependencyFieldUrl === s

    val titleKey =
      if (is(exemptionsAndLossesRoutes.inYearLosses().url)) Some("inYearLosses")
      else if (is(exemptionsAndLossesRoutes.previousYearsLosses().url)) Some("previousYearLosses")
      else if (is(exemptionsAndLossesRoutes.annualExemptAmount().url)) Some("annualExemptAmount")
      else if (is(ytdRoutes.estimatedIncome().url)) Some("income")
      else if (is(ytdRoutes.personalAllowance().url)) Some("personalAllowance")
      else if (is(ytdRoutes.taxableGainOrLoss().url)) Some("taxableGainOrLoss")
      else if (is(ytdRoutes.yearToDateLiability().url)) Some("yearToDateLiability")
      else if (is(ytdRoutes.taxDue().url)) Some("taxOwed")
      else if (is(ytdRoutes.nonCalculatedEnterTaxDue().url)) Some("taxOwed")
      else if (is(ytdRoutes.repayment().url)) Some("repayment")
      else if (is(ytdRoutes.hasEstimatedDetails().url)) Some("hasEstimated")
      else if (is(initialGainorLossRoutes.enterInitialGainOrLoss().url)) Some("initialGainOrLoss")
      else None

    titleKey.map(k => s"unmetDependency.title.$k")
  }
```
it's passing these strings - are they not suitable for inferring what the user is trying to do here?

Reply:
it'll pass things like `unmetDependency.title.personalAllowance` - the `map` at the bottom adds things to the question identifier